git/git.go

```diff
@@ -81,15 +81,27 @@ func CurrentBranch() (string, error) {
 }
 
 func CurrentRemoteRef() (*Ref, error) {
-	remote, err := CurrentRemote()
+	remoteref, err := RemoteRefNameForCurrentBranch()
 	if err != nil {
 		return nil, err
 	}
 
-	return ResolveRef(remote)
+	return ResolveRef(remoteref)
 }
 
-func CurrentRemote() (string, error) {
+// RemoteForCurrentBranch returns the name of the remote that the current branch is tracking
+func RemoteForCurrentBranch() (string, error) {
+	ref, err := RemoteRefNameForCurrentBranch()
+	if err != nil {
+		return "", err
+	}
+	// ref name is full name, split first part for remote
+	parts := strings.Split(ref, "/")
+	return parts[0], nil
```

Comment:
Yeah I know, but that doesn't matter for this? I'm only pulling out the remote name which can't have a '/'.

You can't just append the current branch to the remote name to get the remote tracking ref, that's what the code was doing before and it doesn't deal with tracking differently named remote branches. I fixed that at the same time.